### ISSUE TITLE
aspnet/core/fundamentals/startup "Application Startup in ASP.NET Core"

### ISSUE THREAD

**Issue by Rick-Anderson** [aspnet/Docs#3411]
- [x] Update for 2.0
- [x]  #1307
- [x]  Startup.ConfigureContainer #3659
- [x] Remove all code as sentences.
- [x] remove "from the default web site template"
- [x] Services Available in Startup - rewrite with code showing this. Tell them why you would inject logger in ctor (you want it sooner)
- [x] format code so it fits on tablet screens/no horiz scrolll bars
![image](https://cloud.githubusercontent.com/assets/3605364/26431251/196222ae-408f-11e7-836e-d5b7e8e70348.png)
- [ ]

**Comment by skynode** [comment]
Since this is still open, I thought I'd ask something along these lines to be clear:

I have the following code in the `Startup.cs` for a Web API running on ASP.NET 2.0 (actually migrating from 1.x):

```cs
public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
{
      loggerFactory.AddConsole(Configuration.GetSection("Logging"));
      loggerFactory.AddDebug();
      app.UseMvc();
 }
```
 And this in `Program.cs`:
```cs
    public class Program
    {
        public static IWebHost BuildWebHost(string[] args) =>
            WebHost.CreateDefaultBuilder(args)
                .UseStartup<Startup>()
                .Build();
        public static void Main(string[] args) 
        {
            var host = BuildWebHost(args);
            using(var scope = host.Services.CreateScope())
            {
                var services = scope.ServiceProvider;
                try
                {
                    //Seed the database
                    Seed.Initialize(services.GetRequiredService<MyDbContext>());
                }
                catch (Exception e)
                {
                    var logger = services.GetRequiredService<ILogger<Program>>();
                    logger.LogError(e, $"An error: {e.Message} occurred while initializing the database");
                }
            }
            host.Run();
        }
    }
```

Please, I need to understand how DI works for logging in this scenario; it all compiles but is it runtime correct? I don't have `services.AddLogging()` in the `Startup.cs` container (`IServiceCollection`). Do I need it?

Thanks!

**Comment by Rick-Anderson** [comment]
All fixed except for #5161
